### ISSUE TITLE
Migrations Bundle with Custom Design Time Assembly

### ISSUE THREAD

**Issue by pinkfloydx33** [dotnet/efcore#28551]
Is it possible to customize the dependencies of the migration bundle application? It automatically includes a hard-coded reference to `Microsoft.EntityFrameworkCore.Design` plus the database/startup projects and publishes with their dependencies (direct and transitive). In most cases this should be sufficient but I'm hitting a bit of an issue. 

We currently maintain a lot of customizations to EF to support Postgres row-level-security (model building, migrations, annotations, etc). 

We also have some design time services to support this. We take a *full* reference on `Microsoft.EntityFrameworkCore.Design` and add implementations of `ICsharpSnapshotGenerator`, `IMigrationsCodeGenerator`, `ICsharpMigrationOperationGenerator` and `IAnnotationCodeGenerator`. Our nuget package includes a props file that adds a `DesignTimeServicesReferenceAttribute` to any consuming projects so that the design time services are automatically available. 

Up until now this hasn't been an issue. Everything works great. However, we would like to split out these design time services and create a seperate "Design" development dependency so that we A. aren't flowing the `Microsoft.EntityFrameworkCore.Design` package to our actual applications and B. keep deployed size down even further by not including *our* design time stuff either. 

We took a stab at doing this, using Ms.Ef.Design as a template. We were able to add a reference to the new project, setting `PrivateAssets=all` to prevent flowing transitive deps past our context project. The package itself has a "full" reference to `Microsoft.EntityFrameworkCore.Design` so it's brought in transitively up to this point. 

## The problem

This works fine locally and when running tools such as `Add-Migration`. The bundling process itself succeeds, however attempting to *execute* the bundled application throws a `FileNotFoundException` since our Design Time assembly doesn't get included in the bundle due to PrivateAssets (which I verified with DotPeek, all other transitive refs *are* included). 

So my question here is whether or not it's possible to somehow indicate to the bundling process that it needs to include my extra assembly (without flowing it to other projects)? MS.Ef.Design doesn't suffer here as its literally hard coded into the generated project file. I realize this is an advanced scenario so I'm not going to hold my breath, but I wonder how actual providers would be expected to work if they too had their own design time package (npgsql does not). 

If I can't get it bundled directly, I'm open to alternative such as custom msbuild targets or similar so long as my package doesn't get included with the actual application. I tried various Include/ExcludeAssets values and tried a few custom targets (though I've since lost my efforts so can't share).

Any thoughts? I'm really hoping there's a way through this, but if not then I'll just keep it the way it is now as a combined project.

I realize it may be difficult to comment without seeing exactly what we're doing. Its a giant, internal project so I'm crossing my fingers someone can at least tell me whether I'm on or off track. If needed I can try to put together a really pared down repro. 

### Include provider and version information

EF Core version: 6.0.7
Database provider: npgsql 
Target framework: Net6.0
Operating system: windows 10 x64, Linux-x64 (docker) 
IDE: VS 2022 latest

**Comment by ErikEJ** [comment]
You need to tweak the build process to suit your needs.

**Comment by pinkfloydx33** [comment]
Normal publish/app works fine which is what confuses me. Is it possible that it's due to *unconditionally* adding the `DesignTimeServicesReferenceAttribute` (via `AssemblyAttribute` Msbuild item, added via props file bundled with nuget)?

I just found [this](https://github.com/dotnet/efcore/blob/8144d50ec0d15b87c35a9a3a425b1faadafb87f1/src/EFCore.SqlServer.NTS/build/net6.0/Microsoft.EntityFrameworkCore.SqlServer.NetTopologySuite.targets) which looks like it only conditionally adds the attribute, before CoreCompile and only when `DesignTimeBuild` isn't true

I'll give that a shot. I swear I searched for references to the attribute. This looks promising
